libflash/platformfs.cpp

```diff
@@ -0,0 +1,168 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include "platformfs.h"
+
+#include <unistd.h>
+#include <linux/fs.h>
+#include <fcntl.h>
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+
+#include <sstream>
+
+#include <sys/sysmacros.h>
+#include <mtd/ubi-user.h>
+
+mender::io::ExpectedFile mender::io::Open(const string &p, bool read, bool write) {
+	if (!read && !write) {
+		return Error(std::error_condition(std::errc::io_error), "Wrong access flags provided");
+	}
+	int flags = read && write ? O_RDWR | O_CREAT : write ? O_WRONLY | O_CREAT : O_RDONLY;
```

Comment:
I think we can omit `O_CREAT`, since the tool is always expected to operate on existing device files. This way we can catch mistakes like misspelling the device name, since with `O_CREAT` it would silently create a regular file `/dev` instead.

Reply:
OK, I was using it for testing but we don't need it in the final version.